src/services/legal-services.ts

```diff
@@ -1,76 +1,87 @@
+   * @param tempId the temp registration number
+   * @returns a promise to return the draft filing, else exception
    */
-  static async fetchDraftApplication (id: string): Promise<IncorporationFilingIF | RegistrationFilingIF> {
-    const url = `businesses/${id}/filings`
+  static async fetchFirstOrOnlyFiling (tempId: string): Promise<IncorporationFilingIF | RegistrationFilingIF> {
+    const url = `businesses/${tempId}/filings`
     return axios.get(url)
       .then(response => {
         let filing
         // if response is a list then look at only the first filing
         if (Array.isArray(response?.data?.filings)) filing = response.data.filings[0].filing
         // otherwise expect response to be a single filing
         if (response?.data?.filing) filing = response.data.filing
-        const filingName = filing?.header?.name
+
+        const filingName = filing?.header?.name as FilingTypes
         const filingId = +filing?.header?.filingId || 0
 
         if (!filing || !filingName || !filingId) {
           throw new Error('Invalid API response')
         }
 
-        switch (filingName) {
-          case FilingTypes.INCORPORATION_APPLICATION:
-            return this.formatEmptyIncorporationApplication(filing)
-          case FilingTypes.REGISTRATION:
-            return this.formatEmptyRegistration(filing)
-          default:
-            throw new Error('Invalid filing name')
-        }
+        return filing
       })
   }
 
   /**
-   * Fetches a draft dissolution filing.
-   * @param id the business id
-   * @returns a promise to return the draft dissolution filing, else exception
+   * Fetches the first task.
+   * This is expected to be a draft Dissolution or Restoration.
+   * @param businessId the business identifier
+   * @returns a promise to return the draft filing, else exception
    */
-  static async fetchDraftDissolution (id: string): Promise<DissolutionFilingIF> {
-    // get the draft filing from the tasks endpoint
-    const url = `businesses/${id}/tasks`
+  // eslint-disable-next-line max-len
+  static async fetchFirstTask (businessId: string): Promise<DissolutionFilingIF | RestorationFilingIF> {
+    const url = `businesses/${businessId}/tasks`
     return axios.get(url)
       .then(response => {
-        // look for dissolution task
-        const filing = response?.data?.tasks
-          // eslint-disable-next-line no-prototype-builtins
-          ?.find(x => x.task.filing.hasOwnProperty(FilingTypes.VOLUNTARY_DISSOLUTION))?.task.filing
-        const filingName = filing?.header?.name
+        const filing = response?.data?.tasks?.[0]?.task.filing
+        const filingName = filing?.header?.name as FilingTypes
         const filingId = +filing?.header?.filingId || 0
 
-        if (!filing || filingName !== FilingTypes.VOLUNTARY_DISSOLUTION || !filingId) {
+        if (!filing || !filingName || !filingId) {
           throw new Error('Invalid API response')
         }
 
         return filing
       })
   }
```

Comment:
Renamed and generalized to fetch draft Dissolution or Restoration filing from `tasks` endpoint.